### PR TITLE
Edit Site: Extract gutenberg_find_template_post helper, use in edit-site-page

### PR DESCRIPTION
**Pull request by ockham**
## Description
Extract a `gutenberg_find_template_post` helper function from `gutenberg_find_template`, and rename the latter to `gutenberg_template_include_filter`. Unlike the original function, `gutenberg_find_template_post` makes no references to globals, and can thus be used in `edit-site-page.php` without resorting to as many globals as before. (I've previously found the code in `edit-site-page.php` somewhat hard to read, since it wasn't immediately clear where those globals where coming from.)

This is a preparatory step to better encapsulate the logic that determines the relevant template post for a given page, so it can be more easily reused. Specifically, it's meant as a preparation for implementing an endpoint flag that would make the `wp_template` collection endpoint return only relevant templates, as discussed [here](https://github.com/WordPress/gutenberg/pull/21877/files#r416069139).

(In a follow-up, I might try to get rid of some other `$_wp_current_template_*` globals.)

## How has this been tested?
Verify that Full Site Editing still works as before. Specifically, verify that the template switcher still holds the same templates as on `master`, and that loading them (both front-end and editor) still works.

## Types of changes
Small refactor.

## Checklist:
- [x] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: lib/edit-site-page.php [hunk 1]
```diff
@@ -171,30 +168,30 @@ function gutenberg_edit_site_init( $hook ) {
...
+	$current_template_post = gutenberg_find_template_post( $_wp_current_template_hierarchy );
+	$template_ids[ $current_template_post->post_name ] = $current_template_post->ID;
 	if ( isset( $_wp_current_template_part_ids ) ) {
 		$template_part_ids = $template_part_ids + $_wp_current_template_part_ids;
 	}
-	$settings['templateId']      = $_wp_current_template_id;
+	$settings['templateId']      = $current_template_post->ID;
 	$settings['templateType']    = 'wp_template';
 	$settings['templateIds']     = array_values( $template_ids );
+	$settings['templateIdsAll']  = $template_ids;
```
Comment (epiqueras) [inline lib/edit-site-page.php:194]: What is this used for?

File: lib/template-loader.php [hunk 2]
```diff
@@ -159,18 +159,45 @@ function create_auto_draft_for_template_part_block( $block ) {
...
+	add_action( 'wp_head', 'gutenberg_render_title_tag', 1 );
+
+	// This file will be included instead of the theme's template file.
+	return gutenberg_dir_path() . 'lib/template-canvas.php';
+}
+
+/**
+ * Return the correct 'wp_template' post for the current template hierarchy.
+ *
+ * @param  string[]     $slugs  The current template hierarchy, ordered by priority.
```
Comment (epiqueras) [inline lib/template-loader.php:190]:
```suggestion
 * @param string[] $template_hierarchy The current template hierarchy, ordered by priority.
```

File: lib/template-loader.php [hunk 3]
```diff
@@ -159,18 +159,45 @@ function create_auto_draft_for_template_part_block( $block ) {
...
+ * @return WP_Post|null         A template post object, or null if none could be found.
```
Comment (epiqueras) [inline lib/template-loader.php:191]:
```suggestion
 * @return WP_Post|null A template post object, or null if none could be found.
```

**Review by epiqueras — COMMENTED** [review]
Looks good. I just have one minor question.

Also, as a follow-up, we should also return template part IDs from the new function.

File: lib/edit-site-page.php (hunk 1 above)
Comment (ockham) [reply lib/edit-site-page.php:194]: Oops, debugging leftover. I'll rip it out.

**Comment by ockham** [comment]
> Looks good. I just have one minor question.
> 
> Also, as a follow-up, we should also return template part IDs from the new function.

Yep, planning to work on that :+1:

**Comment by ockham** [comment]
I addressed all feedback -- could I get approval so I can merge? :smile:

**Review by epiqueras — APPROVED** [review]
LGTM

**Pull request merged**